src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_copy_b_emitter.cpp

```diff
@@ -130,107 +83,49 @@ void jit_brgemm_copy_b_emitter::validate_arguments(const std::vector<size_t> &in
-    size_t start_comp = m_comp_offset;
-
-    // OneDNN requires tail handling before main iterations
-    if (m_inner_N_tail != 0) {
-        emit_kernel_call(m_kernel.get(), src, dst, comp, m_inner_N_tail, m_K_blk, start_in, start_out, start_comp);
-        start_in += m_transpose ? m_K * m_inner_N_tail * data_size : m_inner_N_tail * data_size;
-        start_out += m_inner_N_tail * m_brgemmVNNIFactor * data_size;
-        start_comp += m_inner_N_tail * sizeof(int32_t);
-    }
-
-    const size_t in_ld = m_transpose ? m_K * m_inner_N_block * data_size : m_inner_N_block * data_size;
-    const size_t out_ld = m_inner_N_block * m_brgemmVNNIFactor * data_size;
-    const size_t comp_ld = m_inner_N_block * sizeof(int32_t);
-    for (size_t nb = 0; nb < m_N_blk / m_inner_N_block; nb++) {
-        const size_t offset_in = start_in + nb * in_ld;
-        const size_t offset_out = start_out + nb * out_ld;
-        const size_t offset_comp = m_with_comp ? start_comp + nb * comp_ld : 0;
-        emit_kernel_call(m_kernel.get(), src, dst, comp, m_inner_N_block, m_K_blk, offset_in, offset_out, offset_comp);
-    }
+    std::vector<size_t> mem_ptrs_idxs{in[0], out[0]};
+    if (out.size() > 1)
+        mem_ptrs_idxs.emplace_back(out[1]);
+    emit_kernel_call(mem_ptrs_idxs, m_memory_offsets);
 }
 
-void jit_brgemm_copy_b_emitter::emit_kernel_call(const matmul::jit_brgemm_matmul_copy_b_t* kernel, Reg64 src, Reg64 dst, Reg64 comp,
-                                                 size_t N, size_t K, size_t offset_in, size_t offset_out, size_t offset_comp) const {
-    const auto data_ptr = [&](Xmm xmm, Xbyak::Reg64 reg, size_t bytes_offset) {
-        h->uni_vmovq(reg, xmm);
-        if (bytes_offset) h->add(reg, bytes_offset);
+void jit_brgemm_copy_b_emitter::emit_kernel_call(const std::vector<size_t>& mem_ptrs_idxs, const std::vector<size_t>& mem_offsets) const {
+    JitSafeInternalCall safe_internal_caller(h);
+
+    h->mov(h->rbp, reinterpret_cast<uint64_t>(BrgemmCopyBKernelExecutor::execute));
+    auto reserved_stack_size = sizeof(BrgemmCopyBKernel::call_args);
+    // Reserve memory on the stack
+    h->sub(h->rsp, reserved_stack_size);
+
+    std::vector<size_t> used_gpr_idxs = mem_ptrs_idxs;
+    // abi_param1 - runtime parameter register in the kernel which is used for AMX Tile config initialization
+    used_gpr_idxs.push_back(static_cast<size_t>(abi_param1.getIdx()));
+    Xbyak::Reg64 aux_reg = ov::intel_cpu::utils::get_aux_gpr(used_gpr_idxs);
+
+    auto write_addr_on_stack = [&](size_t arg_offset, Reg64 addr, size_t addr_offset, size_t buffer_id) {
+        const auto stack_frame = h->qword[h->rsp + arg_offset];
+        h->mov(aux_reg, addr);
+        if (snippets::utils::is_dynamic_value(addr_offset))
+            h->add(aux_reg,  h->ptr[abi_param1 + GET_OFF(buffer_offsets) + buffer_id * sizeof(size_t)]);
+        else if (addr_offset != 0)
+            h->add(aux_reg, addr_offset);
+        h->mov(stack_frame, aux_reg);
```

Comment:
I think your idea is better: the best place for `write_addr_on_stack` is `JitSafeInternalCall` class. Could you please place the helper there?
I am sorry for not precise suggestion :)

Reply:
To be honest I'm not sure that it's good idea to move it to there for now because `write_addr_on_stack` work with snippets entities (dynamic value check, runtime `abi_param1`) while `JitSafeInternalCall` is common helper in plugin part.
Because of that I moved it to snippets helpers for now.
Also probably we need to discuss it offline 🤔